metricflow/sql_clients/big_query.py

```diff
@@ -57,29 +67,39 @@ def from_connection_details(url: str, password: Optional[str]) -> BigQuerySqlCli
 
         return BigQuerySqlClient(password=password)
 
-    def __init__(self, project_id: str = "", password: Optional[str] = None, query_string: str = "") -> None:
+    def __init__(self, project_id: str = "", password: Optional[str] = None) -> None:
         """Creates a new BigQueryDBClient and tags it for tracing as big query."""
         # Without pool_pre_ping, it's possible for timed-out connections to be returned to the client and cause errors.
         # However, this can cause increase latency for slow engines.
         self._password = password
-        self._project_id = project_id
-        super().__init__(
-            engine=BigQuerySqlClient._create_bq_engine(
-                query_string=query_string, project_id=project_id, password=password
-            )
+        password_json = json.loads(password) if password else {}
+        self._project_id = project_id or password_json.get("project_id")
+        bq_engine = BigQuerySqlClient._create_bq_engine(project_id=project_id, password=password)
+        self._bq_client = bigquery.Client(
+            project=self._project_id,
+            credentials=google.oauth2.service_account.Credentials.from_service_account_info(password_json),
         )
+        super().__init__(engine=bq_engine)
 
     def _engine_specific_dry_run_implementation(self, stmt: str, bind_params: SqlBindParameters) -> None:
         """Overrides base `_engine_specific_dry_run_implementation` function for BigQuery specifics"""
         _engine = self._create_bq_engine(
-            query_string="dry_run=true", project_id=self._project_id, password=self._password
+            query_field_values={"dry_run": "true"}, project_id=self._project_id, password=self._password
         )
         with _engine.connect() as conn:
             conn.execute(sqlalchemy.text(stmt), bind_params.param_dict)
 
+    @staticmethod
+    def _create_bq_url(project_id: str = "", query_field_values: Optional[Dict[str, str]] = None) -> str:
+        if query_field_values is None:
+            query_field_values = {}
+        query_items = tuple(f"{field}:{value}" for field, value in query_field_values.items())
```

Comment:
Wait, `{field}:{value}` not `{field}={value}`?

Reply:
That's a mistake - I was looking at the syntax for labels in the URL. Fixed.